scenarios/validation_receipts/src/main.rs

```diff
@@ -79,6 +79,16 @@ fn agent_behaviour(
     let start = Instant::now();
 
     'outer: loop {
+        // check time out (twice the scenario duration)
+        if ctx
+            .scenario_duration()
+            .map(|d| d * 2)
+            .map(|max_duration| start.elapsed() >= max_duration)
+            .unwrap_or_default()
+        {
+            panic!("Timed out waiting for validation receipts to complete");
```

Comment:
I think we discussed this during refinement and unfortunately it looks like we didn't update the issue with the outcome of that discussion. My recollection is that we said we didn't want to expose the scenario duration in context, but instead refactor the scenario to remove the loop.

One option for that refactoring could be to check at the top if `pending_actions` contains any actions, and if so, check for validation receipts. If we have all the validation receipts, remove it from `pending_actions`. If `pending_actions` is empty, create another entry.